static/js/widget-app.js

```diff
@@ -1,91 +1,114 @@
-window.codesy = {
-};
-
-function submitForm(e) {
-  e.preventDefault()
-  form = $(this)
-  var csrf_token = form.find('input[name=csrfmiddlewaretoken]').val()
-
-  var api_call = $.ajax({
-    url: form.attr('action'),
-    method: form.data("method"),
-    data: form.serialize(),
-    processData: false,
-    headers: {
-      'X-CSRFToken': csrf_token
-    },
-    contentType: 'application/x-www-form-urlencoded; charset=UTF-8'
-
-  })
-
-  api_call.error(function(data, textStatus, jqXHR) {
-    $.each(data.responseJSON, function(key, value) {
-      $('#codesy_widget').prepend(
-        '<div class="callout warning expanded" data-closable>' +
-          '<button class="close-button" data-close>&times;</button>' +
-          '<p class="alert alert-error">' + value + '</p>' +
-        '</div>'
-      )
-    });
-  })
+window.codesy={}
+class WidgetApp {
+    constructor($form) {
+        this.$form = $form
+        $form.find('input[type=text]').keypress(this.ignore_period)
+        $form.find('input#submitForm').submit(this.submit.bind(this))
+        $form.find('button#ShowSubmit').click(this.show.bind(this))
+        $form.find('input#cancelSubmit').click(this.reload.bind(this))
+    }
+    show(e) {
+        e.preventDefault()
+        $('.codesy_hide').hide()
```

Comment:
We should change this class name to be semantic to the content, not the function. Maybe change it to `submit-element` (as compared to `confirm-element`s ?